ambry-replication/src/main/java/com/github/ambry/replication/PartitionLeaderInfo.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.replication;
+
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.clustermap.ReplicaState;
+import com.github.ambry.server.StoreManager;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+
+
+/**
+ * Maintains the list of leader partitions on local node and their corresponding peer leaders in remote data centers
+ */
+public class PartitionLeaderInfo {
+
+  private final Map<String, Set<ReplicaId>> peerLeaderReplicasByPartition;
+  protected final StoreManager storeManager;
+  private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+  public PartitionLeaderInfo(StoreManager storeManager) {
+    peerLeaderReplicasByPartition = new ConcurrentHashMap<>();
+    this.storeManager = storeManager;
+  }
+
+  /**
+   * Get a map of partitions to their sets of peer leader replicas
+   * @return an unmodifiable map of peer leader replicas stored by partition {@link PartitionLeaderInfo#peerLeaderReplicasByPartition}
+   */
+  public Map<String, Set<ReplicaId>> getPeerLeaderReplicasByPartition() {
+    return Collections.unmodifiableMap(peerLeaderReplicasByPartition);
```

Comment:
I assume this will called by replica thread. My question is how frequently it is called?  Do we need to get full snapshot of `peerLeaderReplicasByPartition`？